links: use libpng16 instead of libpng.

fbrosson opened the pull request:
* Drop unneeded support for secondary architectures.
* Depend on libssl and libcrypto instead of openssl.
* Use **`$effectiveTargetArchitecture`** (instead of **`$targetArchitecture`**) when defining STDCPPLIBS (as **`-lstdc++`** or *``lbstdc++.r5`**) in order to have a working code for both primary and secondary architectures,
  should that code snippet be copied to other recipes.
* Add TEST() with **`make check`**.

fbrosson commented:
This PR fixes #1101. Tested on x86_gcc2. Both links and links_x86 build and run fine.
I have a few questions:
1) Should I drop support for secondary architectures (and the CONFLICTS I added)?
2) Should I replace the dependency on openssl by a dependency on libssl and/or libcrypto?
3) Should I bump revision?

korli commented:
1. why was support for secondary architectures present in the first place?
2. it depends, does links also build with openssl 1.0.2?
3. yes, this changes the resulting package.

fbrosson commented:
OK, thanks. I've updated the recipe.

1) I I don't know. I've just removed the support for 2ndary arch.
2) Yes, it builds fine with openssl 1.0.2. Dependencies changed to libssl and libcrypto.
3) done.
